test/parallel/test-buffer-compare-offset.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+const a = Buffer.from([1, 2, 3, 4, 5, 6, 7, 8, 9, 0]);
+const b = Buffer.from([5, 6, 7, 8, 9, 0, 1, 2, 3, 4]);
+
+assert.equal(-1, a.compare(b));
+
+// Equivalent to a.compare(b).
+assert.equal(-1, a.compare(b, 0));
+
+// Equivalent to a.compare(b).
+assert.equal(-1, a.compare(b, 0, undefined, 0));
+
+// Zero-length compare, return 0
+assert.equal(0, a.compare(b, 0, 0, 0));
+
+// Equivalent to Buffer.compare(a, b.slice(6, 10))
+assert.equal(1, a.compare(b, 6, 10));
+
+// Zero-length compare, return 0
+assert.equal(0, a.compare(b, 6, 10, 0, 0));
+
+// Equivalent to Buffer.compare(a.slice(4), b.slice(0, 5))
+assert.equal(1, a.compare(b, 0, 5, 4));
+
+// Equivalent to Buffer.compare(a.slice(1), b.slice(5))
+assert.equal(1, a.compare(b, 5, undefined, 1));
+
+// Equivalent to Buffer.compare(a.slice(2), b.slice(2, 4))
+assert.equal(-1, a.compare(b, 2, 4, 2));
+
+// Equivalent to Buffer.compare(a.slice(4), b.slice(0, 7))
+assert.equal(-1, a.compare(b, 0, 7, 4));
+
+// Equivalent to Buffer.compare(a.slice(4, 6), b.slice(0, 7));
+assert.equal(-1, a.compare(b, 0, 7, 4, 6));
+
+// Overlong end offset is tolerated
+assert.equal(-1, a.compare(b, 0, 100, 0));
+
+const oor = /out of range index/;
+const otherstart_re = /"targetStart" argument must be an integer/;
+const thisstart_re = /"sourceStart" argument must be an integer/;
+const thisend_re = /"sourceEnd" argument must be an integer/;
+const otherend_re = /"targetEnd" argument must be an integer/;
+const arg_re = /Argument must be a Buffer/;
+
+assert.throws(() => a.compare(b, -1), oor);
+assert.throws(() => a.compare(b, 0, -1), oor);
+assert.throws(() => a.compare(b, 0, 0, -1), oor);
+assert.throws(() => a.compare(b, 0, 0, 0, -1), oor);
+assert.throws(() => a.compare(), arg_re);
```

Comment:
Can we add tests passing values like `'0xff'`, `null`, `Infinity` and even `{ valueOf: () => 5 }`?